cmd/memtierd/main.go

```diff
@@ -85,18 +134,8 @@ func main() {
 
 	flag.Parse()
 
-	switch *optLog {
-	case "", "stderr":
-		memtier.SetLogger(log.New(os.Stderr, "", 0))
-	case "-", "stdout":
-		memtier.SetLogger(log.New(os.Stdout, "", 0))
-	default:
-		logFile, err := os.OpenFile(*optLog, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
-		if err != nil {
-			exit("failed to open log file %q: %v", *optLog, err)
-		}
-		memtier.SetLogger(log.New(logFile, "", 0))
-	}
+	processLoggerFlag(*optLog)
```

Comment:
As explained in the overall review, I'm slightly against moving almost randomly selected set of command line option handling away from the main and keeping the rest.

Forgetting about the complexity scores (as we are not doing this for the score, we are doing it for the human maintainers), if you think that optLog should be handled outside the main function, consider changing the name of the function descriptive so that it tells what's the main thing it does. In this case it is calling the memtier.SetLogger function. Therefore the function could be `setLogger(*optLog)`, or if you wish to move line 139 in it, too, then `configureLog(*optLog, *optDebug)` to handle all log-related configuration in one place.